src/Compilers/CSharp/Portable/FlowAnalysis/NullableWalker.cs

```diff
@@ -4847,7 +5249,18 @@ public override BoundNode VisitBadExpression(BoundBadExpression node)
         public override BoundNode VisitTypeExpression(BoundTypeExpression node)
         {
             var result = base.VisitTypeExpression(node);
+
+            if (node.BoundContainingTypeOpt != null)
+            {
+                VisitTypeExpression(node.BoundContainingTypeOpt);
+            }
+
+            // VisitTypeExpression is called recursively, so we need to manually set the previous expression before
+            // updating the result;
+            var previousExpression = _currentExpression;
+            _currentExpression = node;
             SetNotNullResult(node);
+            _currentExpression = previousExpression;
```

Comment:
Why does `SetNotNullResult(node)` rely on setting `_currentExpression`?

Reply:
Because setting any result relies on `_currentExpression`. As I'm removing `_currentExpression`, this will become irrelevant.

---
In reply to: [265724208](https://github.com/dotnet/roslyn/pull/34017#discussion_r265724208) [](ancestors = 265724208)